jekyll/_cci2/pipeline-variables.md

```diff
@@ -143,10 +135,15 @@ Configuration processing happens in the following stages:
 
 The remaining configuration is processed, element parameters are resolved, type-checked, and substituted.
 
-## Element parameter scope
+## The scope of pipeline parameters
+{: #the-scope-of-pipeline-parameters }
+
+Pipeline parameters can only be resolved in the `.circleci/config.yml` file in which they are declared. Pipeline parameters are not available in orbs, including orbs declared locally in your `.circleci/config.yml` file. This is because access to pipeline scope in orbs would break encapsulation and create a hard dependency between the orb and the calling configuration. This would potentially jeopardize determinism and create a surface area of vulnerability.
```

Comment:
Is there any chance we could either use more plain language or include / link out to definitions..? I am thinking that "dependency" and "vulnerability" seem OK, but "encapsulation" and "determinism" might be a bit more esoteric... We may be able to simplify "create a surface area of vulnerability" too, e.g. increase security risk, make project/pipeline/app more vulnerable to attacks - something along those lines?

Reply:
@kbatuigas I'm updating the second part, but I think encapsulation is okay - it's a fairly common programming concept.